main.go

```diff
@@ -37,58 +37,6 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
-// After running the executable, you may connect to it using the following:
-//
-// > mysql --host=localhost --port=3306 --user=root
-//
-// The included MySQL client is used in this example, however any MySQL-compatible client will work.
-
-var (
-	initMode = false
-
-	address       = "0.0.0.0"
-	port          = 3306
-	socket        string
-	dataDirectory = "."
-	dbFileName    = "mysql.db"
-	logLevel      = int(logrus.InfoLevel)
-
-	replicaOptions replica.ReplicaOptions
-
-	postgresPort = 5432
-
-	defaultTimeZone = ""
-)
-
-func init() {
-	flag.BoolVar(&initMode, "init", initMode, "Initialize the program and exit. The necessary extensions will be installed.")
-
-	flag.StringVar(&address, "address", address, "The address to bind to.")
-	flag.IntVar(&port, "port", port, "The port to bind to.")
-	flag.StringVar(&socket, "socket", socket, "The Unix domain socket to bind to.")
-	flag.StringVar(&dataDirectory, "datadir", dataDirectory, "The directory to store the database.")
-	flag.IntVar(&logLevel, "loglevel", logLevel, "The log level to use.")
-
-	// The following options need to be set for MySQL Shell's utilities to work properly.
-
-	// https://dev.mysql.com/doc/refman/8.4/en/replication-options-replica.html#sysvar_report_host
-	flag.StringVar(&replicaOptions.ReportHost, "report-host", replicaOptions.ReportHost, "The host name or IP address of the replica to be reported to the source during replica registration.")
-	// https://dev.mysql.com/doc/refman/8.4/en/replication-options-replica.html#sysvar_report_port
-	flag.IntVar(&replicaOptions.ReportPort, "report-port", replicaOptions.ReportPort, "The TCP/IP port number for connecting to the replica, to be reported to the source during replica registration.")
-	// https://dev.mysql.com/doc/refman/8.4/en/replication-options-replica.html#sysvar_report_user
-	flag.StringVar(&replicaOptions.ReportUser, "report-user", replicaOptions.ReportUser, "The account user name of the replica to be reported to the source during replica registration.")
-	// https://dev.mysql.com/doc/refman/8.4/en/replication-options-replica.html#sysvar_report_password
-	flag.StringVar(&replicaOptions.ReportPassword, "report-password", replicaOptions.ReportPassword, "The account password of the replica to be reported to the source during replica registration.")
-
-	// The following options are used to configure the Postgres server.
-
-	flag.IntVar(&postgresPort, "pg-port", postgresPort, "The port to bind to for PostgreSQL wire protocol.")
-
-	// The following options configure default DuckDB settings.
-
-	flag.StringVar(&defaultTimeZone, "default-time-zone", defaultTimeZone, "The default time zone to use.")
-}
-
```

Comment:
Could we revert this change? While a global configuration package might seem convenient initially, it can lead to long-term maintainability issues in Go projects. A global configuration introduces implicit dependencies, making it harder to understand how different parts of the application are connected and how data flows. Currently, main.go controls the flow by explicitly passing configurations to modules, which offers several benefits: 1) It makes modules independently testable by directly injecting settings without touching global configuration; and 2) it provides a transparent view of what each module requires. Long-term, we may want to use a flag parsing library like [xo/ox](https://github.com/xo/ox) for improved parsing, however, we will still explicitly pass configuration to each module to maintain control of data flow and avoid global implicit dependencies.

Reply:
Sure If it is not a good practice, I'll revert it.

Reply:
fixed